Update functionality matrix page for NF ver 24.04.2 (#74)
seqeralabs/docs: platform_versioned_docs/version-23.4.0/functionality_matrix/functionality_matrix.mdx

Reviewer: Be aware that changes were pushed to the table for versions `23.3.0` and `23.4.0`. You'll need to update your table accordingly.

Reviewer 2: These should be pulled in automatically, thanks! :)

tags: [compatibility, nextflow, nf-launcher]
---

Each Seqera Platform version makes use of `nf-launcher` to determine the Nextflow version used as its baseline. You can override this version with the `NXF_VER` environment variable during pipeline launch, but note that Seqera may not work reliably with Nextflow versions other than the baseline version.

> Seqera officially supports the two latest major Platform releases (23.4.x, 23.3.x, etc.) at any given time.

Reviewer 2: ```suggestion
The two most recent major versions (23.4.x, 23.3.x, etc.) of Seqera Platform are supported at any given time.
```

tags: [compatibility, nextflow, nf-launcher]
---

Each Seqera Platform version makes use of `nf-launcher` to determine the Nextflow version used as its baseline. You can override this version with the `NXF_VER` environment variable during pipeline launch, but note that Seqera may not work reliably with Nextflow versions other than the baseline version.

> The two most recent major versions (23.4.x, 23.3.x, etc.) of Seqera Platform are supported at any given time.

Reviewer: I assume you'll update this again come 24.1 release?

| Platform version | nf-launcher version | Nextflow version | Fusion version |
| ---------------- | ------------------- | ---------------- | -------------- |
| *                |                     | 24.04.2          | 2.3            |
| 23.4.4           | j17-23.10.1         | 23.10.1          | 2.2            |
| 23.4.3           | j17-23.10.1         | 23.10.1          | 2.2            |
| 23.4.2           | j17-23.10.1         | 23.10.1          | 2.2            |
| 23.4.1           | j17-23.10.1         | 23.10.1          | 2.2            |
| 23.3.0           | j17-23.10.0         | 23.10.0          | 2.1            |
| 23.2.0           | j17.23.04.2-up3     | 23.04.2          | 2.1            |
| 23.1.3           | j17-23.04.1         | 23.04.1          | 2.1            |

:::note 
Nextflow edge versions after 24.04.2 use Fusion 2.4 by default. 
:::

---

> To use a Nextflow version other than the baseline in your pipeline run, use a [pre-run script](./launch/advanced#pre-and-post-run-scripts) during launch. For example, enter `export NXF_VER=24.04.2` in the **Pre-run script** field to take advantage of [new Nextflow features](https://nextflow.io/blog/2024/nextflow-2404-highlights.html) before a Platform version is released with 24.04.2 as its baseline. 

Reviewer: You've got NXF_VER stuff above the table too. Does it make sense to have this split across the page?

Reviewer: Maybe keep the NXF_VER stuff light here (i..e just mention you can overrride) and then link to the `launch/advanced.mdx` page for actual details how to do it.

Author: Simplified.

> Each Seqera Platform version makes use of `nf-launcher` to determine the Nextflow version used as its baseline. This Nextflow version can be overridden with the `NXF_VER` environment variable in your `nextflow.config` file, but note that Seqera may not work reliably with Nextflow versions other than the baseline version.

Reviewer 3: ```suggestion
Each Seqera Platform version makes use of `nf-launcher` to determine the Nextflow version used as its baseline.
```

Duplicated in the section below.

It's also wrong - can't set this environment variable within a `nextflow.config` file, or at least it has no effect as Nextflow has already launched by the time the config file is parsed.

| 23.3.0           | j17-23.04.3         | 23.04.3          | 2.1            |
| 23.2.0           | j17.23.04.2-up3     | 23.04.2          | 2.1            |
| 23.1.3           | j17-23.04.1         | 23.04.1          | 2.1            |

(* - Nextflow version 24.04.2 will be the baseline version of an upcoming Platform release, but you can take advantage of [new Nextflow features](https://nextflow.io/blog/2024/nextflow-2404-highlights.html) in current Platform versions by specifying `NXF_VER=24.04.2` in your configuration.)

---

> To use a Nextflow version other than the baseline in your pipeline run, include the environment variable `NXF_VER` in a [pre-run script](./launch/advanced#pre-and-post-run-scripts) during launch. For example, `NXF_VER=24.04.2`. 

Reviewer: Do we need to call out that post 24.04.x edge versions use Fusion v2.4 by default?

| 23.3.0           | j17-23.04.3         | 23.04.3          | 2.1            |
| 23.2.0           | j17.23.04.2-up3     | 23.04.2          | 2.1            |
| 23.1.3           | j17-23.04.1         | 23.04.1          | 2.1            |

:::note 
Nextflow edge versions later than 24.04.2 use Fusion 2.4 by default. 
:::

> (* - Nextflow version 24.04.2 will be the baseline version of an upcoming Platform release, but you can take advantage of [new Nextflow features](https://nextflow.io/blog/2024/nextflow-2404-highlights.html) in current Platform versions by specifying `NXF_VER=24.04.2` in your configuration.)

Reviewer 4: Per Google Developer Style, it is recommended to avoid hinting at future state. Maybe 24.04.17 becomes the baseline? I don't know that we've actually adopted this prohibition, though, but wanted to mention.

Reviewer 2: That's a good point, especially as it sets an expectation that may be delayed or rolled back etc.

Reviewer 3: Agree on removing this - it also seems to essentially duplicate the bit below about how to change versions.

> (* - Nextflow version 24.04.2 will be the baseline version of an upcoming Platform release, but you can take advantage of [new Nextflow features](https://nextflow.io/blog/2024/nextflow-2404-highlights.html) in current Platform versions by specifying `NXF_VER=24.04.2` in your configuration.)

Reviewer 2: When we say "current Platform versions," are those all versions?

---

> To use a Nextflow version other than the baseline in your pipeline run, include the environment variable `NXF_VER` in a [pre-run script](./launch/advanced#pre-and-post-run-scripts) during launch. For example, `NXF_VER=24.04.2`. 

Reviewer: Shouldnt we include instructions on how to swap Fusion versions too? (it uses a similar mechanism as Nextflow version swapping).

Reviewer 3: Is this documented anywhere? I can't find info on how to do this.
